### PR TITLE
Add SpacyEmbeddings class

### PR DESCRIPTION
**Pull request by ghost**
- Description: Added a new SpacyEmbeddings class for generating embeddings using the Spacy library.
- Issue: Sentencebert/Bert/Spacy/Doc2vec embedding support #6952
- Dependencies: This change requires the Spacy library and the 'en_core_web_sm' Spacy model.
- Tag maintainer: @dev2049
- Twitter handle: N/A

This change includes a new SpacyEmbeddings class, but does not include a test or an example notebook.

### CODE REVIEW

File: langchain/embeddings/spacy_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        """
+        Asynchronously generates embeddings for a list of documents.
+
+        Args:
+            texts (List[str]): The documents to generate embeddings for.
+
+        Returns:
+            A list of embeddings, one for each document.
+        """
+        return self.embed_documents(texts)
```
Comment (baskaryan) [inline langchain/embeddings/spacy_embeddings.py:97]: we shouldn't use the synchronous methods here. can just raise a NotImplementedError if we don't want to implement async right now

**Comment by baskaryan** [comment]
nice! could we add an example notebook showing how to use this? probably would live in docs/extras/modules/data_connection/text_embedding/integrations

**Comment by rjarun8** [comment]
> nice! could we add an example notebook showing how to use this? probably would live in docs/extras/modules/data_connection/text_embedding/integrations

Sure, i will get one created

File: langchain/embeddings/spacy_embeddings.py (hunk 1 above)
Comment (rjarun8) [reply langchain/embeddings/spacy_embeddings.py:97]: I need some more explanation to understand the comment better. Can you pls share some reference design pattern or interface templates to be followed for implementing embedding integrations. @baskaryan

Comment (rjarun8) [reply langchain/embeddings/spacy_embeddings.py]: > we shouldn't use the synchronous methods here. can just raise a NotImplementedError if we don't want to implement async right now

@baskaryan I have made the changes and committed.

https://github.com/hwchase17/langchain/pull/6967/commits/52b0cc4be90527ca01fcd9cf9b63120b6f35df24

File: docs/extras/modules/data_connection/text_embedding/spacy_embedding.ipynb [hunk 2]
```diff
@@ -0,0 +1,86 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the necessary classes\n",
```
Comment (hwchase17) [inline docs/extras/modules/data_connection/text_embedding/spacy_embedding.ipynb:9]: notebooks should have a title of `# Spacy` and then all other cells should NOT have `#` but should be more like `## Import the ....`

File: langchain/embeddings/spacy_embeddings.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+import importlib.util
+from typing import Any, Dict, List, Optional
+from pydantic import BaseModel, Extra, root_validator
+from langchain.embeddings.base import Embeddings
+
+
+class SpacyEmbeddings(BaseModel, Embeddings):
```
Comment (hwchase17) [inline langchain/embeddings/spacy_embeddings.py:7]: lets expose this in __init__ of embeddings module as well

**Review by hwchase17 — COMMENTED** [review]
this is great - thanks!

File: docs/extras/modules/data_connection/text_embedding/spacy_embedding.ipynb (hunk 2 above)
Comment (rjarun8) [reply docs/extras/modules/data_connection/text_embedding/spacy_embedding.ipynb]: > notebooks should have a title of `# Spacy` and then all other cells should NOT have `#` but should be more like `## Import the ....`

done. Notebook updated

File: langchain/embeddings/spacy_embeddings.py (hunk 3 above)
Comment (rjarun8) [reply langchain/embeddings/spacy_embeddings.py:7]: > lets expose this in **init** of embeddings module as well

done. exposed in init.

**Comment by baskaryan** [comment]
thanks @rjarun8!

**Pull request merged**